dio/lib/src/dio_mixin.dart

```diff
@@ -359,19 +359,16 @@ abstract class DioMixin implements Dio {
     ProgressCallback? onSendProgress,
     ProgressCallback? onReceiveProgress,
   }) async {
-    options ??= Options();
-    final requestOptions = options.compose(
+    final requestOptions = (options ??= Options()).compose(
       this.options,
       path,
       data: data,
       queryParameters: queryParameters,
       onReceiveProgress: onReceiveProgress,
       onSendProgress: onSendProgress,
       cancelToken: cancelToken,
+      sourceStackTrace: StackTrace.current,
```

Comment:
This is the actual stack trace that matters in order to find out where the client code initiated the request.
After this point inside `fetch` there are async gaps and then the stack trace will look somewhat like this:

```
  File "dio_mixin.dart", line 643, in DioMixin.fetch.<fn>
  File "zone.dart", line 1422, in _rootRunBinary
  File "zone.dart", line 1314, in _CustomZone.runBinary
  File "future_impl.dart", line 162, in _FutureListener.handleError
  File "future_impl.dart", line 779, in Future._propagateToListeners.handleError
  File "future_impl.dart", line 800, in Future._propagateToListeners
  File "future_impl.dart", line 575, in Future._completeError
  File "future_impl.dart", line 51, in _SyncCompleter._completeError
  File "future_impl.dart", line 23, in _Completer.completeError
  File "future.dart", line 617, in Future.any.onError
  File "zone.dart", line 1422, in _rootRunBinary
  File "zone.dart", line 1314, in _CustomZone.runBinary
  File "future_impl.dart", line 162, in _FutureListener.handleError
  File "future_impl.dart", line 779, in Future._propagateToListeners.handleError
  File "future_impl.dart", line 800, in Future._propagateToListeners
  File "future_impl.dart", line 575, in Future._completeError
  File "future_impl.dart", line 520, in Future._chainForeignFuture.<fn>
  File "zone.dart", line 1398, in _rootRun
  File "zone.dart", line 1300, in _CustomZone.run
  File "zone.dart", line 1208, in _CustomZone.runGuarded
  File "zone.dart", line 1248, in _CustomZone.bindCallbackGuarded.<fn>
  File "schedule_microtask.dart", line 40, in _microtaskLoop
  File "schedule_microtask.dart", line 49, in _startMicrotaskLoop
  ```
  
  And you have no idea where it came from.